spec/unit/resource/api500/c7000/volume_spec.rb

```diff
@@ -87,19 +87,31 @@
   end
 
   describe '#set_storage_pool' do
-    it 'sets the storagePool attribute' do
-      item = described_class.new(@client_500)
+    it 'sets the storagePool attribute in properties' do
+      item = described_class.new(@client_500, properties: {})
       item.set_storage_pool(OneviewSDK::StoragePool.new(@client_500, uri: '/rest/fake'))
       expect(item['properties']['storagePool']).to eq('/rest/fake')
     end
+
+    it 'should not sets the storagePoolUri without properties' do
+      item = described_class.new(@client_500)
+      item.set_storage_pool(OneviewSDK::StoragePool.new(@client_500, uri: '/rest/fake'))
+      expect(item['storagePoolUri']).to_not be
+    end
   end
 
   describe '#set_snapshot_pool' do
-    it 'sets the snapshotPool attribute' do
-      item = described_class.new(@client_500)
+    it 'sets the snapshotPool attribute in properties' do
+      item = described_class.new(@client_500, properties: {})
```

Comment:
if the user are not pass the properties as an empty `Hash` and set the storate_pool, will the create action create a Volume with that storage pool associated? If not, is this behavior correct?
For me, no problem use this new approach, but maybe important documenting that over the constructor of the Volume class.